### PR TITLE
PR for Issue #134 Unitary Hack

### PR DESCRIPTION
**Pull request by bopardikarsoham**
Hello, @Hanrui-Wang. I added a sample PR to the codebase by adding the SDG gate. I intend to add more, but first I'd like you to check if I did everything right. As described in Issue #134, I put the sdg matrix to the matrix dictionary in functional.py and wrote a function for the gate. Next, I added the sdg class to the operators.py file, and finally, I added the sdg gate to the test_op.py file.

### CODE REVIEW

File: torchquantum/functional.py [hunk 1]
```diff
@@ -1168,6 +1169,9 @@ def singleexcitation_matrix(params):
     ),
     "ecr": torch.tensor(
         [[0, 0, 1, 1j], [0, 0, 1j, 1], [1, -1j, 0, 0], [-1j, 1, 0, 0]], dtype=C_DTYPE
+    ),
+    "sdg": torch.tensor(
+        [[1 + 0j, 0 + 0j],[0 + 0j, -0 - 1j]], dtype=C_DTYPE
```
Comment (Hanrui-Wang) [inline torchquantum/functional.py:1174]: Could you change this to [[1, 0], [0, -1j]] for better readability?

File: torchquantum/functional.py [hunk 2]
```diff
@@ -3220,6 +3224,52 @@ def ecr(
...
+    q_device: QuantumDevice,
+    wires: Union[List[int], int],
+    params: torch.Tensor = None,
+    n_wires: int = None,
+    static: bool = False,
+    parent_graph=None,
+    inverse: bool = False,
+    comp_method: str = "bmm",
+):
+    """Perform the hadamard gate.
```
Comment (Hanrui-Wang) [inline torchquantum/functional.py:3237]: here should be the sdg gate

File: torchquantum/functional.py (hunk 1 above)
Comment (bopardikarsoham) [reply torchquantum/functional.py:1174]: Sure!

File: torchquantum/functional.py (hunk 2 above)
Comment (bopardikarsoham) [reply torchquantum/functional.py:3237]: Yes, sorry

**Comment by bopardikarsoham** [comment]
I've made the necessary modifications @Hanrui-Wang ; does the current PR appear to be satisfactory? Should I keep adding gates in this format?

**Comment by bopardikarsoham** [comment]
Hello @Hanrui-Wang, I have added 3 more gates TDG, SXDG and CH.

**Comment by Hanrui-Wang** [comment]
Hi Soham, the added gates looks good, thank you for your contributions! could you pull request to the dev branch?

**Comment by bopardikarsoham** [comment]
Sure @Hanrui-Wang , I'll PR to the dev branch

**Comment by bopardikarsoham** [comment]
I changed the base branch to dev, but I think some of the previous commits from the repo also got added to my PR

**Comment by bopardikarsoham** [comment]
Hi @Hanrui-Wang just updating you regarding my progress,

Non-Parameterized Gates:

- [x] SDG
- [x] TDG
- [x] SXDG
- [x] CH
- [x] CCZ
- [x] iSWAP
- [x] CS
- [x] CSDG
- [x] CSX

Parameterized Gates:

- [x] R

I believe I have covered the majority of the missing gates, but I'll continue to look for additional ones.

Further Work:

Adding more ansatz layers(EfficientSU2, RealAmplitudes, etc.) and NLocal circuits.

**Comment by Hanrui-Wang** [comment]
Thank you @bopardikarsoham for your contributions!

**Pull request merged**